### DISCUSSION TITLE
Better playground /w captures

### DISCUSSION THREAD

**Discussion by emilebosch** [tree-sitter/tree-sitter#2164] [Ideas] [answered]
Would be nice to have a better playground that has more language support and also supports captures. Tree parsing is super awesome and now that ASTS become more unified and normalised would be nice if there was a better playground around. Would also encourage future devs to play, and maybe be less intimidated

As of now learning new the queries takes a bit of effort 😅. More samples would also help.

**Comment by ahelwer** [comment, marked as answer]
The playground does support captures (click the query checkbox) and you can run the playground for any language by downloading the grammar and running tree-sitter playground in the root of the repo (need specific version of emscripten though).

**Reply by emilebosch** [reply]
Thank you for answering.
Maybe I miss something. But I can't see the actual result of the captures. Not in the same manner as the CLI does. Yes, I can click the query but it doesn't spit out the capture results.

The idea of a playground would for me more be to fiddle with queries until I got it right. Much like the online regex tools that you have, that serve both as an learning and a experimentation tool. It would help to lower the barrier to people working with queries, and building code navigation systems.

**Reply by ahelwer** [reply]
The captures are highlighted in the editor pane, although not in the main demo playground at https://tree-sitter.github.io/tree-sitter/playground (it's unclear what grammar versions are being used anyway).

If you run tree-sitter playground they will be. You can see an example of this here: https://tlaplus-community.github.io/tree-sitter-tlaplus/

**Reply by sogaiu** [reply]
Nice.

I gave it a try too.

![tree-sitter-playground-with-query-highlighting](https://user-images.githubusercontent.com/33044872/229387624-33e77bec-2091-43da-94d8-8b9520593f4d.png)

**Reply by emilebosch** [reply]
Im sorry but there seem to be some confusion, since this doesn't solve the problem I am having,  as what that playground does is having TLA and just hightligting but I go beyond that for parsing and docs generation. What I'm suggesting is:

A more general playground with:
- capture support. I..e the same way you can see the output when you do `tree-sitter query` with ranges, the patterns and the matches
- a bunch of languages to experiment with (basically everything that is now a package under `tree-sitter-x` packages)
- many examples 
- node type definitions

Reason for this is that when building tools (like I'm right now building a general outline/doc tools) I need to write a ton of queries for various languages. Such a tool would help.

**Reply by ahelwer** [reply]
Well, this is a volunteer-driven project so if you would like to see these features integrated with the main playground at https://tree-sitter.github.io/tree-sitter/playground you can certainly consider opening a PR! The topic of what languages to include has come up previously in this issue: https://github.com/tree-sitter/tree-sitter/issues/1094
